lib/ttfunk/table/hmtx.rb

```diff
@@ -22,15 +22,20 @@ def self.encode(hmtx, mapping)
       HorizontalMetric = Struct.new(:advance_width, :left_side_bearing)
 
       def for(glyph_id)
-        @metrics[glyph_id] ||
-          HorizontalMetric.new(
+        metrics_cache[glyph_id] ||= begin
+          @metrics[glyph_id] || HorizontalMetric.new(
```

Comment:
There's no need to cache the value that are readily available.

Try this:
```ruby
@metrics[glyph_id] || metrics_cache[glyph_id] ||= 
  HorizontalMetric.new(...)
```

Reply:
I'm trying to avoid creating `HorizontalMetric` instances when not necessary. In the cases where the metric is already available then yes, it will be cached by `metrics_cache`. In the case when it's not available, we create it and store it. I suppose we could do something like

```ruby
@metrics[glyph_id] || (metrics_cache[glyph_id] ||= HorizontalMetric.new(...))
```

Is that what you would prefer?

Reply:
Ah ok cool.